stockpile-wrapper/stockpile-wrapper.py

```diff
@@ -45,26 +46,35 @@ def _index_result(server,port,my_uuid,my_node,my_pod,es_ssl):
     stockpile_file = os.popen('grep stockpile_output_path group_vars/all.yml | awk \'{printf $2}\'').read()
 
     if os.path.exists(stockpile_file):
-        _upload_to_es(stockpile_file,my_uuid,timestamp,es,index,my_node,my_pod)
-        _upload_to_es_bulk(stockpile_file,my_uuid,timestamp,es,index,my_node,my_pod)
+        _upload_to_es(stockpile_file, my_uuid, timestamp, es, my_node, my_pod)
+        _upload_to_es_bulk(stockpile_file, my_uuid, timestamp, es, index, my_node, my_pod)
 
-def _upload_to_es(payload_file,my_uuid,timestamp,es,index,my_node,my_pod):
+def _upload_to_es(payload_file, my_uuid, timestamp, es, my_node, my_pod):
+    indexed = 0
+    failed = 0
     payload = open(payload_file, "rb").read()
-    for scribed in transcribe(payload_file,'stockpile'):
-        try:
-            scribe_module = json.loads(scribed)['module']
-            _data = { "uuid": my_uuid,
-                            "timestamp": timestamp, 
-                            "node_name": my_node, 
-                            "pod_name": my_pod, }
-            _data.update(json.loads(scribed))
-            es.index(index=scribe_module+"-metadata", body=_data)
-        except Exception as e:
-            print(repr(e) + "occurred for the json document:")
-            print(str(scribed))
-            indexed=False
-
-def _upload_to_es_bulk(payload_file,my_uuid,timestamp,es,index,my_node,my_pod):
+    def doc_stream():
+        for scribed in transcribe(payload_file,'stockpile'):
+            try:
+                doc = json.loads(scribed)
+                scribe_module = doc["module"]
+                data = {"uuid": my_uuid, "timestamp": timestamp, "node_name": my_node, "pod_name": my_pod}
+                data.update(doc)
+                yield {"_index": scribe_module + "-metadata",
+                       "_source": data,
+                       "_op_type": "index"}
+            except Exception as e:
+                print(repr(e) + "occurred for the json document:")
+                print(str(scribed))
+    for ok, resp in parallel_bulk(es, doc_stream()):
+        if ok:
+            indexed += 1
+        else:
+            failed += 1
+    print("%d documents indexed sucessfully. %d documents could't be indexed" % (indexed, failed))
```

Comment:
Seems like we should do something about the failed documents? Other than just alerting? ie, maybe add a retry, and write to disk if the retry fails? Write the json payload.

Reply:
Sure, the current logic doesn't have retry, but that's an improvement in case of getting a 409 from ES.